design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+The diagram shows the result of the following sequence of actions:
+
+1. The user creates an `StackInstall` and specifies the WordPress stack
+1. The Stack Manager installs the CRDs defined in the stack, creates the necessary RBAC primitives, and starts up the WordPress controller in the same namespace as the stack.
+1. The user creates an instance of the `WordPressInstance` custom resource in the same namespace.
+1. The WordPress controller is watching for events on this CRD type in the namespace it is running in.
+1. In response to seeing the creation event for this `WordPressInstance` custom resource, the WordPress controller creates claims for the services that the instance will need, such as a MySQL database and a Kubernetes cluster.
+    1. Note: in order to have permissions to create instances of MySQL and Kubernetes cluster claims, the Wordpress stack will have to have declared those CRD types as dependencies.
+1. These claims are then serviced by the default resource classes in the environment that point to specific provisioners that were previously installed as part of a cluster stack.
+
+This flow demonstrates how a stack indirectly consumes a cluster stack and is represented in the diagram below:
+
+![Diagram of a Stack Installed](./images/stack-installed.png)
+
+### Multiple Versions
+
+As previously mentioned, since stacks will be installed per namespace, multiple instances of each stack can be installed in the control plane.
+This also means that multiple versions of the stack can be installed side by side.
+Because of the namespaced isolation, this is fine for the stack's controllers, but it does cause a potential difficulty for the CRDs.
+
+The custom resource **definitions** themselves are always cluster scoped.
+If two versions of a stack are installed in the control plane and they have identical CRD versions, there could be a conflict.
+To illustrate this clearly, consider the following example:
+
+1. Stack `foo-v1.0` is installed along with its CRD that has version `v1beta1`
+1. A minor update to stack foo is released as version `foo-v1.1`
+    1. This update to the stack adds a new field to the CRD but does **not** increment the CRD version
+1. The control plane now has 2 versions of the stack's controllers (`foo-v1.0` and `foo-v1.1`), but only 1 version of the CRD (`v1beta`)
+
+Which CRD schema should "win" in the control plane?
+
+* The **new CRD schema** with the new field that the **old controller** doesn't understand?
+* The **old CRD schema** without the new field that the **new controller** wants to use?
+
+This speaks to the importance of properly versioning CRDs when they have schema changes.
+Kubernetes now supports [multiple side by side versions of CRDs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions) in the cluster, so this is actually possible in more recent versions of Kubernetes.
```

Comment:
"Multiple side by side versions of CRDs" reads a little to me like Kubernetes supports installing multiple distinct, versioned, instances of a particular CRD kind rather than the reality in which on CRD kind may define multiple versions.

Reply:
I'll clarify the wording around multiple versions, and that the experience is still cumbersome.